src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_INDEX = 2;
+    private Node<K, V> [] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        int threshold = (int) (table.length * LOAD_FACTOR);
+        if (size > threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private void resize() {
+        Node<K, V>[] lastBuckets = table;
+        int newCapacity = table.length * RESIZE_INDEX;
+        size = 0;
+        table = new Node[newCapacity];
+        for (Node<K, V> node : lastBuckets) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private static class Node<K, V> {
```

Comment:
![image](https://github.com/mate-academy/jv-hash-map/assets/90136583/82fa64af-e9b5-4fd8-9910-119564d587b8)
Maybe you don't have all fields.

Reply:
I don't need the hash value in my implementation, the hash value is used internally within the getIndex(K key) method to determine the index